src/format.js

```diff
@@ -1,15 +1,27 @@
-import { get, template } from 'lodash'
+import { get } from 'lodash'
+import IntlMessageFormat from 'intl-messageformat'
+import defaultLocaleData from './en'
+import { addLocaleData } from './utils'
 
-const interpolate = /{{([\s\S]+?)}}/g
 const defaultLabel = {}
+const browserLanguage = window.navigator.language || window.navigator.browserLanguage
+/* add default locale on startup, import on src folder */
+addLocaleData(defaultLocaleData)
 
-export default locale => (config, componentProps) =>
+export const formatter =
+  (lang, label = '', customFormat) => new IntlMessageFormat(label, lang, customFormat)
+
+export default (locale, lang = browserLanguage, customFormats) => (config, componentProps) =>
   Object
     .keys(config)
     .reduce((acc, curr) => {
       const label = get(locale, config[curr])
-      if (componentProps && componentProps[curr]) {
-        return { ...acc, [curr]: template(label, { interpolate })(componentProps[curr]) }
+      if (label && componentProps && componentProps[curr]) {
+        /* use formatJS to parse message */
+        return {
+          ...acc,
+          [curr]: formatter(lang, label, customFormats).format(componentProps[curr]),
```

Comment:
I think the purpose of formatter() (before `format`) is to cache all formatter.
So calls to `format` are fast.

There is something to do here I guess, can you create an issue ?

Reply:
I was wondering, but the constructor of `IntlMessageFormat` take the `message` and it is mandatory.

Do you have any idea to cache this ?

Reply:
Create an issue please to argue about this

Reply:
#7